### PR TITLE
bpo-29688: Document Path.absolute

### PR DESCRIPTION
**Pull request by DimitrisJim**

### CODE REVIEW

**Review by marco-buttu — CHANGES_REQUESTED** [review]
LGTM. You should just make the code markup on `'.'` (\`\`'.'\`\`) and `'..'` (\`\`'..'\`\`), as in the rest of the doc.

**Comment by DimitrisJim** [comment]
Good catch, changed it. Thanks, Marco!

**Review by marco-buttu — APPROVED** [review]

File: Doc/library/pathlib.rst [hunk 1]
```diff
@@ -652,6 +652,18 @@ call fails (for example because the path doesn't exist):
    .. versionadded:: 3.5
 
 
+.. method:: Path.absolute()
+
+   Return an absolute version of this path.  This function works
```
Comment (brettcannon) [inline Doc/library/pathlib.rst]: Maybe rephrase as "Return an absolute version of this path based on the current working directory"?

File: Doc/library/pathlib.rst [hunk 2]
```diff
@@ -652,6 +652,18 @@ call fails (for example because the path doesn't exist):
...
+   even if the path doesn't point to anything.
```
Comment (brettcannon) [inline Doc/library/pathlib.rst]: "even if the path doesn't exist".

File: Doc/library/pathlib.rst [hunk 3]
```diff
@@ -652,6 +652,18 @@ call fails (for example because the path doesn't exist):
...
+
+   No normalization is done, i.e all '.' and '..' will be kept along.
```
Comment (brettcannon) [inline Doc/library/pathlib.rst]: "will be kept along" -> "will not be normalized".

**Review by brettcannon — CHANGES_REQUESTED** [review]

**Comment by DimitrisJim** [comment]
Thanks @brettcannon, made the changes you requested.

**Review by marco-buttu — COMMENTED** [review]
Much better than the method docstring :-) Thanks @brettcannon :-)

**Review by brettcannon — APPROVED** [review]

**Comment by brettcannon** [comment]
Thanks! When I have time to also do the cherry-pick PRs I will merge.

File: Doc/library/pathlib.rst [hunk 4]
```diff
@@ -652,6 +652,18 @@ call fails (for example because the path doesn't exist):
    .. versionadded:: 3.5
 
 
+.. method:: Path.absolute()
+
+   Return an absolute version of this path based on the current working
+   directory.  This function works even if the path doesn't exist.
```
Comment (vstinner) [inline Doc/library/pathlib.rst:658]: This function => This method

I dislike "works even if", maybe: "This method doesn't check if the path exists." ?

File: Doc/library/pathlib.rst [hunk 5]
```diff
@@ -652,6 +652,18 @@ call fails (for example because the path doesn't exist):
    .. versionadded:: 3.5
 
 
+.. method:: Path.absolute()
+
+   Return an absolute version of this path based on the current working
```
Comment (vstinner) [inline Doc/library/pathlib.rst:657]: Hum, maybe specify that only relative paths are resolved using the current working directory?

Return an absolute version of this path. Relative paths are resolved using the current working directory.

**Review by vstinner — CHANGES_REQUESTED** [review]

**Review by berkerpeksag — COMMENTED** [review]
Note that the `absolute` method is barely tested in `test_pathlib`. https://github.com/python/cpython/blob/master/Lib/test/test_pathlib.py#L1343 is the only place it's indirectly tested.

There is also the following comment in its implementation:

```py
# FIXME this must defer to the specific flavour (and, under Windows,
# use nt._getfullpathname())
```

https://github.com/python/cpython/blob/master/Lib/pathlib.py#L1096

**Comment by vstinner** [comment]
Oh wow, these comments are scary. I would feel more confortable with a change adding more tests and removing "# FIXME this must defer to the specific flavour (and, under Windows, use nt._getfullpathname())" comment.

**Comment by brettcannon** [comment]
That would help explain why it isn't documented. 😉 

@DimitrisJim are you up for writing some tests for the function?

**Comment by DimitrisJim** [comment]
I'd rather leave the testing for someone more aware of the intricacies of paths for different OS's.  Should we close the PR and rename the issue on b.p.o to something like "pathlib.absolute needs tests and documentation"?

**Comment by brettcannon** [comment]
Sure, we can close this for now and re-open if we decide to not deprecate Path.absolute(). Thanks for the PR regardless of whether it gets merged.
